Fixes bugs caused by datum mutations

Razharas opened the pull request:
Fixes visibility in crit
Fixed xray and nightvision not working
Fixed dna changing diseases
Fixed hulks doing damage on all intents
Fixed transform procs sometimes breaking dna forever

Razharas committed:
General fixes of bugs caused by mutations

Fixes visibility in crit
Fixed xray and nightvision not working
Fixed dna changing diseases
Fixed hulks doing damage on all intents
Fixed transform procs sometimes breaking dna forever

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into GenFixes

RemieRichards commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -27,7 +27,6 @@
 	internal_organs += new /obj/item/organ/brain
 
 	// for spawned humans; overwritten by other code
-	create_dna(src)
```
Why?

Razharas replied:
ready_nda includes creade_dna

RemieRichards replied:
Eh I guess it does. nevermind then.

tkdrg commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -94,7 +96,7 @@
 
 	//Status updates, death etc.
-	handle_regular_status_updates()		//TODO: optimise ~Carn
+		//TODO: optimise ~Carn
```
Move the comments too, please

Razharas replied:
To be honest id better delete all them comments but whatever

tkdrg commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -462,7 +453,7 @@
 
 		//Eyes
-		if(disabilities & BLIND)	//disabled-blind, doesn't get better on its own
+		if(disabilities & BLIND || stat)	//disabled-blind, doesn't get better on its own
```
Doesn't this mean that living mobs will never heal from temporary blindness? Why'd you do this?

Ergovisavi replied:
stat == 1 is unconscious, stat == 2 is dead. It's possible to recover from it by sleep, apparently?

tkdrg replied:
Oh, I misread that. Yeah, you're right, my bad.

tkdrg commented on code/modules/mob/transform_procs.dm:
```diff
@@ -143,18 +150,14 @@
-			O.real_name = random_name(O.gender)
+		if(cmptext("monkey",copytext(O.dna.real_name,1,7)))
+			O.dna.real_name = random_name(O.gender)
+	O.real_name = O.dna.real_name
```
This needs to be intended one more level, otherwise you're overwriting newname.

Razharas replied:
Ill just move it above newname overwrite

tkdrg commented on code/modules/mob/transform_procs.dm:
```diff
@@ -143,18 +150,14 @@
 	O.name = O.real_name
 
-	if (!(tr_flags & TR_KEEPSE))
-		O.dna.struc_enzymes = setblock(O.dna.struc_enzymes, RACEBLOCK, construct_block(1,BAD_MUTATION_DIFFICULTY))
```
Why are you removing this?

Razharas replied:
It is moved

Razharas committed:
Fixes some bugs and rutimes tkdrg pointed

Monkeys arent blind anymore
Bad dna mutation runtime fixed
Two handed shit runtime fixed
Carbons unarmed attacks runtime fixed

Razharas committed:
Last pack of fixes for now

Fixed indentation in transform proc
Moved comments to appropriate place
Fixed unreported cloning bug
Removed monkifying from random bad mutations

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into GenFixes
